File: 1390-four-divisors/1390-four-divisors.py
class Solution:
    def sumFourDivisors(self, nums: List[int]) -> int:
        ans = 0
        for num in nums:
            summ = 0
            count = 0

            n = 1
            while n*n <= num:
                # if its divisible by n == divisor
                if num % n == 0:
                    count += 1
                    summ += n

                    d = num // n
                    if d != n:
                        count += 1
                        summ += d
                    
                    if count > 4:
                        break
                n += 1
            
            if count == 4:
                ans += summ
        return ans

File: 1390-four-divisors/1390-four-divisors.cpp
class Solution {
public:
    int sumFourDivisors(vector<int>& nums) {
        int sum = 0;

        for (int num : nums) {
            int divisorCount = 0;
            int divisorSum = 0;

            for (int i = 1; i <= sqrt(num); i++) {
                if (num % i == 0) {
                    divisorCount++;
                    divisorSum += i;
                    if (num / i != i) {
                        divisorCount++;
                        divisorSum += num / i;
                    }
                }
                if (divisorCount > 4) {
                    break;
                }
            }

            if (divisorCount == 4) {
                sum += divisorSum;
            }
        }

        return sum;
    }
};
